File: selenium_junit5_parallel_with_wdm/src/test/java/org/o2f/pages/automationpractice/components/HeaderComponent.java
package org.o2f.pages.automationpractice.components;

import org.o2f.pages.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class HeaderComponent extends BasePage {

    @FindBy(css = "#header .banner .img-responsive")
    private WebElement headerDiscounts;

    @FindBy(css = "[class='logo img-responsive']")
    private WebElement logo;

    @FindBy(name = "submit_search")
    private WebElement submitSearchBtn;

    @FindBy(name = "search_query")
    private WebElement searchInputBox;

    @FindBy(css = "a.login")
    private WebElement loginBtn;

    public HeaderComponent(WebDriver driver) {
        super(driver);
    }

    public void headerDiscountShouldBeDisplayed() {
        wait.until((driver1) -> headerDiscounts.isDisplayed());
    }

    public void logoShouldBeDisplayed() {
        wait.until(ExpectedConditions.visibilityOf(logo));
    }

    public void loginBtnShouldBeDisplayed() {
        wait.until(driver1 -> loginBtn.isDisplayed());
    }

    public void search(String text) {
        inputText(searchInputBox, text);
        clickOnBtn(submitSearchBtn);
    }

}

File: python-selenium-pure/pages/automation_practice/components/header_component.py
from selenium.webdriver.common.by import By

from pages.base_page import BasePage


class HeaderComponent(BasePage):

    header_discounts: By = (By.CSS_SELECTOR, "#header .banner .img-responsive")
    logo: By = (By.CSS_SELECTOR, "[class='logo img-responsive']")
    submit_search_btn: By = (By.NAME, "submit_search")
    search_input_box: By = (By.NAME, "search_query")
    login_btn: By = (By.CSS_SELECTOR, "a.login")

    def __init__(self, driver):
        super().__init__(driver)

    def header_discount_should_be_displayed(self):
        element = self.wait_until_element_displayed(self.header_discounts)
        assert element is not None or element != ""

    def login_btn_should_be_displayed(self):
        self.wait_until_element_displayed(self.login_btn)

    def search(self, text: str):
        self.input_text(self.search_input_box, text)
        self.click_on_btn(self.submit_search_btn)

